lib/rspec/mocks/error_generator.rb

```diff
@@ -62,6 +62,18 @@ def raise_unexpected_message_args_error(expectation, *args)
       end
 
       # @private
+      def raise_unexpected_message_with_args_error(expectation, *args)
+        expected_args = format_args(*expectation.expected_args)
+        actual_args = format_unexpected_with_args(*args)
+        diff = diff_message(expectation.expected_args, args)
+
+        message = default_error_message(expectation, expected_args, actual_args)
+        message << "\nDiff:#{diff}" unless diff.empty?
+
+        __raise message
+      end
```

Comment:
Was definitely trying to take the path of least resistance by keeping with the same convention without making too many changes to existing methods.  The new method was because of the variance of arguments that were coming in made it difficult to standardize the response. If it's okay to adjust existing callers to use a new combined  method, then it should open up an easier path to make it into one method as you suggest.